cogito/putter.go

```diff
@@ -105,51 +122,63 @@ func (putter *ProdPutter) ProcessInputDir() error {
 		}
 	}
 
-	if inputDirs.Size() == 0 {
-		return fmt.Errorf(
-			"put:inputs: missing directory for GitHub repo: have: %v, GitHub: %s/%s",
-			collected, source.Owner, source.Repo)
-	} else if inputDirs.Size() > 1 {
+	switch inputDirs.Size() {
+	case 0:
+		// If the size is 0 after removing the directory containing the chat message
+		// and Cogito should update the commit status, return an error.
+		if sets.From(sinks...).Contains("github") {
+			return fmt.Errorf(
+				"put:inputs: missing directory for GitHub repo: have: %v, GitHub: %s/%s",
+				inputDirs, source.Owner, source.Repo)
+		}
+		putter.log.Debug("", "inputDirs", inputDirs, "msgDir", msgDir)
+	case 1:
+		repoDir := filepath.Join(putter.InputDir, inputDirs.OrderedList()[0])
+		putter.log.Debug("", "inputDirs", inputDirs, "repoDir", repoDir, "msgDir", msgDir)
+		if err := checkGitRepoDir(repoDir, source.Owner, source.Repo); err != nil {
+			return err
+		}
+		putter.gitRef, err = getGitCommit(repoDir)
+		if err != nil {
+			return err
+		}
+		putter.log.Debug("", "git-ref", putter.gitRef)
+	default:
+		// If the size exceeds 1, too many directories are passed to Cogito.
 		return fmt.Errorf(
 			"put:inputs: want only directory for GitHub repo: have: %v, GitHub: %s/%s",
 			inputDirs, source.Owner, source.Repo)
 	}
 
-	// The set has one or two elements. if it exists, remove from the set the message
-	// directory. The remaining one is the git repo.
-	putter.log.Debug("", "inputDirs", inputDirs, "msgDir", msgDir)
-	remaining := inputDirs.Difference(sets.From(msgDir))
-	repoDir := filepath.Join(putter.InputDir, remaining.OrderedList()[0])
-
-	if err := checkGitRepoDir(repoDir, source.Owner, source.Repo); err != nil {
-		return err
-	}
-
-	putter.gitRef, err = getGitCommit(repoDir)
-	if err != nil {
-		return err
-	}
-	putter.log.Debug("", "git-ref", putter.gitRef)
-
 	return nil
 }
 
 func (putter *ProdPutter) Sinks() []Sinker {
-	return []Sinker{
-		GitHubCommitStatusSink{
+	supportedSinkers := map[string]Sinker{
+		"github": GitHubCommitStatusSink{
 			Log:     putter.log.Named("ghCommitStatus"),
 			GhAPI:   putter.ghAPI,
 			GitRef:  putter.gitRef,
 			Request: putter.Request,
 		},
-		GoogleChatSink{
+		"gchat": GoogleChatSink{
 			Log: putter.log.Named("gChat"),
 			// TODO putter.InputDir itself should be of type fs.FS.
 			InputDir: os.DirFS(putter.InputDir),
 			GitRef:   putter.gitRef,
 			Request:  putter.Request,
 		},
 	}
+	source := putter.Request.Source.Sinks
+	params := putter.Request.Params.Sinks
+	sinks, _ := MergeAndValidateSinks(source, params)
+
+	sinkers := make([]Sinker, 0, len(sinks))
+	for _, s := range sinks {
```

Comment:
With `MergeAndValidateSinks` returning a Set this would just becomes:
```suggestion
	sinkers := make([]Sinker, 0, sinks.Size())
	for _, s := range sinks.OrderedList() {
```

Reply:
fixed